tests/test_keras_mixin.py

```diff
@@ -0,0 +1,153 @@
+            return self.l1(x)
+
+
+else:
+    DummyModel = None
+
+
+@require_tf
+class HubMixingCommonTest(unittest.TestCase):
+    _api = HfApi(endpoint=ENDPOINT_STAGING)
+
+
+@require_tf
+class HubMixingTest(HubMixingCommonTest):
+    def tearDown(self) -> None:
+        try:
+            shutil.rmtree(WORKING_REPO_DIR, onerror=set_write_permission_and_retry)
+        except FileNotFoundError:
+            pass
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Share this valid token in all tests below.
+        """
+        cls._token = cls._api.login(username=USER, password=PASS)
+
+    def test_save_pretrained(self):
+        model = DummyModel()
+
+        model.save_pretrained(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        self.assertTrue("tf_model.h5" in files)
+        self.assertEqual(len(files), 1)
+
+        model.save_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}", config={"num": 12, "act": "gelu"}
+        )
+        files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        self.assertTrue("config.json" in files)
+        self.assertTrue("tf_model.h5" in files)
+        self.assertEqual(len(files), 2)
+
+    def test_keras_from_pretrained_weights(self):
+        model = DummyModel()
+        model.dummy_inputs = None
+        model.save_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}", dummy_inputs=tf.ones([2, 2])
+        )
+        assert model.built
+        new_model = DummyModel.from_pretrained(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        # Check the reloaded model's weights match the original model's weights
+        self.assertTrue(tf.reduce_all(tf.equal(new_model.weights[0], model.weights[0])))
+
+        # Check a new model's weights are not the same as the reloaded model's weights
+        another_model = DummyModel()
+        another_model(tf.ones([2, 2]))
+        self.assertFalse(
+            tf.reduce_all(tf.equal(new_model.weights[0], another_model.weights[0]))
+            .numpy()
+            .item()
+        )
+
+    def test_rel_path_from_pretrained(self):
+        model = DummyModel()
+        model.save_pretrained(
+            f"tests/{WORKING_REPO_SUBDIR}/FROM_PRETRAINED",
+            config={"num": 10, "act": "gelu_fast"},
+        )
+
+        model = DummyModel.from_pretrained(
+            f"tests/{WORKING_REPO_SUBDIR}/FROM_PRETRAINED"
+        )
+        self.assertTrue(model.config == {"num": 10, "act": "gelu_fast"})
+
+    def test_abs_path_from_pretrained(self):
+        model = DummyModel()
+        model.save_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}-FROM_PRETRAINED",
+            config={"num": 10, "act": "gelu_fast"},
+        )
+
+        model = DummyModel.from_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}-FROM_PRETRAINED"
+        )
+        self.assertDictEqual(model.config, {"num": 10, "act": "gelu_fast"})
+
+    # def test_push_to_hub(self):
```

Comment:
Oops! Commented this out so I wasn't flooding the staging endpoint while working. Uncomment before merging.